docs/content/config/advanced/kubernetes.md

```diff
@@ -4,518 +4,626 @@ title: 'Advanced | Kubernetes'
+        # imaps (implicit TLS)
+        - name: imaps
+          port: 993
+          targetPort: imaps
+          protocol: TCP
+    ```
+
+=== "`Deployment`"
+
+    Last but not least, the `Deployment` becomes the most complex component. It instructs Kubernetes how to run the DMS container and how to apply your `ConfigMaps`, persisted storage, etc. Additionally, we can set options to enforce runtime security here.
+
+    ```yaml
+    ---
+    apiVersion: apps/v1
+    kind: Deployment
+
+    metadata:
+      name: mailserver
+
       annotations:
-        container.apparmor.security.beta.kubernetes.io/mailserver: runtime/default
+        ignore-check.kube-linter.io/run-as-non-root: >-
+          'mailserver' needs to run as root
+        ignore-check.kube-linter.io/privileged-ports: >-
+          'mailserver' needs privileged ports
+        ignore-check.kube-linter.io/no-read-only-root-fs: >-
+          There are too many files written to make The
+          root FS read-only
 
     spec:
-      hostname: mail
-      containers:
-        - name: mailserver
-          image: ghcr.io/docker-mailserver/docker-mailserver:latest
-          imagePullPolicy: IfNotPresent
-
-          securityContext:
-            # Required to support SGID via `postdrop` executable
-            # in `/var/mail-state` for Postfix (maildrop + public dirs):
-            # https://github.com/docker-mailserver/docker-mailserver/pull/3625
-            allowPrivilegeEscalation: true
-            readOnlyRootFilesystem: false
-            runAsUser: 0
-            runAsGroup: 0
-            runAsNonRoot: false
-            privileged: false
-            capabilities:
-              add:
-                # file permission capabilities
-                - CHOWN
-                - FOWNER
-                - MKNOD
-                - SETGID
-                - SETUID
-                - DAC_OVERRIDE
-                # network capabilities
-                - NET_ADMIN  # needed for F2B
-                - NET_RAW    # needed for F2B
-                - NET_BIND_SERVICE
-                # miscellaneous  capabilities
-                - SYS_CHROOT
-                - KILL
-              drop: [ALL]
-            seccompProfile:
-              type: RuntimeDefault
-
-          # You want to tune this to your needs. If you disable ClamAV,
-          #   you can use less RAM and CPU. This becomes important in
-          #   case you're low on resources and Kubernetes refuses to
-          #   schedule new pods.
-          resources:
-            limits:
-              memory: 4Gi
-              cpu: 1500m
-            requests:
-              memory: 2Gi
-              cpu: 600m
-
-          volumeMounts:
+      replicas: 1
+      selector:
+        matchLabels:
+          app: mailserver
+
+      template:
+        metadata:
+          labels:
+            app: mailserver
+
+          annotations:
+            container.apparmor.security.beta.kubernetes.io/mailserver: runtime/default
```

Comment:
No worries as of now, we can safely still have it here. When AppArmor is GA, we can take actions.

Reply:
> When AppArmor is GA, we can take actions.

I don't think it will be GA? The last link I provided is dated, while the earlier ones are quite recent.

https://github.com/kubernetes/kubernetes/pull/123435

> - _converts the AppArmor annotations to SecurityContext fields._
> - _migrate Seccomp annotations to fields:_
> - _AppArmor profiles can now be configured through fields on the PodSecurityContext and container SecurityContext._
> - _The beta AppArmor annotations are deprecated._

---

If you feel it's better to leave here in deprecated form, instead of removing or adapting to the v1.30 `SecurityContext` fields no worries then 👍

Reply:
Absolutely not an issue IMO; deprecations usually provide users with a lot of time, and v1.30 is not even out at this moment.